example/push/Main.hs

```diff
@@ -0,0 +1,52 @@
+module Main where
+
+import Control.Exception (displayException)
+import Data.Aeson ((.=))
+import Data.Monoid ((<>))
+
+import qualified Data.Aeson as A
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.HashMap.Strict as H
+import qualified Data.Text as T
+import qualified Data.Text.IO as T
+import qualified Data.Yaml as Y
+import qualified Network.Pusher as P
+import qualified Network.Pusher.Protocol as P
+
+main :: IO ()
+main = do
+  eitherCred <- Y.decodeFileEither "../credentials.yaml"
+  case eitherCred of
+    Left e -> print e
+    Right cred -> do
+      pusher <- P.getPusher cred
+      demoPushNotification pusher
+
+demoPushNotification :: P.Pusher -> IO ()
+demoPushNotification pusher = do
+  let Just interest = P.mkInterest "exampleinterest"
+      fcmObject :: A.Object
+      fcmObject =
+        H.fromList
+          [ ( "notification"
+            , A.Object $
+              H.fromList
+                [ ("title", A.String "Sup?")
+                , ("body", A.String "Hello Firebase.")
+                , ("icon", A.String "firebase-logo.png")
+                ])
+          ]
+      notification =
+        P.Notification
+        { P.notificationInterest = interest
+        , P.notificationWebhookURL = Nothing
+        , P.notificationWebhookLevel = Nothing
+        , P.notificationAPNSPayload = Nothing
+        , P.notificationGCMPayload = Nothing
+        , P.notificationFCMPayload = Just $ P.FCMPayload fcmObject
+        }
```

Comment:
I agree. I've left it for now but there are several possible options here.

- Setter functions
```
notification :: Interest -> Notification
notification i = Notification i Nothing Nothing Nothing Nothing Nothing

setAPNS :: APNSPayload -> Notification -> Notification
```
- And/Or Record selectors
Record selectors could also be encouraged to be used on the values like:

```
(notification "interest"){notificationAPNSPayload = Just ...}
```

- And/Or Lens
Alternatively 'lens's could be used to write code like:

```
notification "interest" & apnsPayload .~ ...
                        & gcmPayload  .~ ...
```

I'm sure there are other points in the design space which all have their tradeoffs.